### PR TITLE
Rewrite some workflow history size calculation

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Use implicit contract, instead of multiple round trip updating history size in 2 different places
  * Persistence layer will keep tract of history size & update history size

**Why?**
N/A

**How did you test it?**
more tests

**Potential risks**
N/A

**Is hotfix candidate?**
N/A

### CODE REVIEW

File: service/history/ndc/conflict_resolver.go [hunk 1]
```diff
@@ -180,10 +180,10 @@ func (r *ConflictResolverImpl) rebuild(
 		return nil, err
 	}
 	rebuildMutableState.GetExecutionInfo().VersionHistories = versionHistories
+	rebuildMutableState.AddHistorySize(rebuiltHistorySize)
```
Comment (yycptt) [inline service/history/ndc/conflict_resolver.go:183]: this is because `rebuildMutableState` initially has history size of 0? so Add and Set are the same?

**Review by yycptt — APPROVED** [review]

**Comment by yycptt** [comment]
Not related to this PR. 
I am always confused by the fact that HistorySize in mutable state is updated by persistence layer by directly changing the executionInfo field in the request. Guess we will eventually have to fix it, if persistence is separate service.

File: service/history/ndc/conflict_resolver.go (hunk 1 above)
Comment (wxing1292) [reply service/history/ndc/conflict_resolver.go:183]: yep

**Comment by wxing1292** [comment]
> Not related to this PR. I am always confused by the fact that HistorySize in mutable state is updated by persistence layer by directly changing the executionInfo field in the request. Guess we will eventually have to fix it, if persistence is separate service.

the history size should not be calculated / updated by the persistence layer, on the surface, DB returnning the size makes sense, but if you take into account how a row is stored, the size makes no sense: 
DB may have multiple full / partial copies of the row -> which size to return? before compaction or after?

in the ideal world
* the history events should be serialized by history builder and persisted to DB as binary
* the history events should not be deserialized & serialized again by frontend when SDK need to access history events for replay
* history size can be calculated & updated to mutable state before calling into persistence layer, so mutable state should not be updated in multiple places ....

**Review by yux0 — APPROVED** [review]

**Pull request merged**